### PR TITLE
Added JComboBox in MassSetFieldDialog

### PR DESCRIPTION
**Pull request by oscargus**
Added a JComboCox for "field" in MassSetFieldDialog (Set/clear/rename field in right-click menu).

Seemed to be quite convenient compared to typing the field name.

### CODE REVIEW

File: src/main/java/net/sf/jabref/util/MassSetFieldAction.java [hunk 1]
```diff
@@ -85,6 +87,18 @@ private void createDialog() {
         set = new JRadioButton(Localization.lang("Set fields"));
         rename = new JRadioButton(Localization.lang("Rename field to") + ":");
         rename.setToolTipText(Localization.lang("Move contents of a field into a field with a different name"));
+        
+        Collection<BibtexEntry> be = frame.basePanel().database().getEntries();
+        allFields = new TreeSet<String>();
+        for (BibtexEntry e : be) {
+            allFields.addAll(e.getAllFields());
+        }
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/util/MassSetFieldAction.java]: Can you extract the logic of getting all the fields of a database and move this method in the database class itself? Maybe use getAllVisibleFields as a method, then you can also ignore the ones that start with "__".

File: src/main/java/net/sf/jabref/util/MassSetFieldAction.java [hunk 2]
```diff
@@ -150,7 +164,7 @@ public void stateChanged(ChangeEvent e) {
             public void actionPerformed(ActionEvent e) {
                 // Check if the user tries to rename multiple fields:
                 if (rename.isSelected()) {
-                    String[] fields = getFieldNames(field.getText());
+                    String[] fields = getFieldNames((String) field.getSelectedItem());
```
Comment (simonharrer) [inline src/main/java/net/sf/jabref/util/MassSetFieldAction.java]: hm, strange that a generically typed combo box requires a cast to String. Is this really necessary?

Comment (koppor) [inline src/main/java/net/sf/jabref/util/MassSetFieldAction.java]: Yeah, it is by design as arbitrary objects can be inserted if the ComboBox is editable: http://stackoverflow.com/a/7026724/873282

Solution: Replace the code by 

    field.getItemAt(field.getSelectedIndex())

(or add a comment pointing to the stackoverflow answer :innocent:)

Comment (simonharrer) [inline src/main/java/net/sf/jabref/util/MassSetFieldAction.java]: Okay, so we can leave it at this.

**Comment by oscargus** [comment]
As you may have noted, I have used the new Eclipse settings which apparently reformats everything when saving...

**Comment by koppor** [comment]
When using git gui, you can choose the lines to commit. Just mark them and right click and   "stage lines to commit". Would that be possible? After committing: Ctrl r and revert the other changes!

----- Ursprüngliche Nachricht -----
Von: "Oscar Gustafsson" <notifications@github.com>
Gesendet: ‎24.‎10.‎2015 20:05
An: "JabRef/jabref" <jabref@noreply.github.com>
Cc: "Oliver Kopp" <koppdev@googlemail.com>
Betreff: Re: [jabref] Added JComboBox in MassSetFieldDialog (#255)

As you may have noted, I have used the new Eclipse settings which apparently reformats everything when saving...

**Comment by simonharrer** [comment]
Hm, the new eclipse settings seem to be wrong. We should revert to the old ones as they cause a lot of havoc at the moment. See the strange indents in this commit.

**Comment by koppor** [comment]
I tried with Eclipse Mars SR1 and it did NOT format it in a wrong way. I checked out this branch and did `CTRL+a` and `CTRL+SHIFT+f` at `BibtexDatabase` and it fixed everything.

@oscargus: which Eclipse version do you use? Mine is as follows

    Version: Mars.1 Release (4.5.1)
    Build id: 20150924-1200

**Comment by oscargus** [comment]
I use the same version. By looking at the code it seemed like it was the `listener` variable that messed things up. I moved it to the end and now it looks much better. A bit strange since `listener` is also "correctly" formatted now.

**Comment by koppor** [comment]
Could you switch the two commits `Move listener in the code ...` and `Added method`? Then we can properly read the diffs.

Side note: I tried to do CTRL+a and CTRL+f in Eclipse. Then it formatted correctly. Only the save actions seem to mess up the code. :cry:

**Comment by oscargus** [comment]
I'll try that. rebase -i, right? Or should I simply just merge the commit
with the strange formatting and the better formatting?

**Comment by koppor** [comment]
`rebase -i` should work. Think, you'll get many conflicts. Maybe you just keep the first commit only (meaning deleting the `pick` lines for the other two commits) and then do the other commits manually again.

Two options for that:
1. Copy the file to a temp folder and copy'n'paste the method during editing
2. Use a [second worktree](https://github.com/blog/2042-git-2-5-including-multiple-worktrees-and-triangular-workflows) and checkout the last commit there
3. Use `gitk --all&` and copy'n'paste the change from there. You have to remove the `-` though.

I personally do mostly option 3, sometimes option 2.

**Comment by simonharrer** [comment]
you can also use `git reset HEAD~2` and then create a new commit.

**Pull request merged**